### PR TITLE
ci: add dynamic release/publish capability

### PR DESCRIPTION
**Pull request by hayemaxi**
- To publish to marketplace.
- Env var `TARGET_EXTENSION` will be provided by the release pipeline.
- Only one .vsix is expected at a time.
- Only 'toolkit' will get the full release for now (e.g. a github release). We will revisit full releasing soon.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildspec/release/20buildrelease.yml [hunk 1]
```diff
@@ -19,13 +20,17 @@ phases:
...
+            - npm run package -w packages/${TARGET_EXTENSION}
+            - cp packages/${TARGET_EXTENSION}/package.json ./package.json
 
 artifacts:
     files:
-        - aws-toolkit-vscode*
-        - packages/toolkit/package.json
-    discard-paths: true
+        - '*.vsix'
+        - package.json
```
Comment (justinmk3) [inline buildspec/release/20buildrelease.yml:36]: don't we want `discard-paths: true` ?

File: buildspec/release/50githubrelease.yml [hunk 2]
```diff
@@ -19,22 +19,24 @@ phases:
...
 
     build:
         commands:
+            - echo "TARGET_EXTENSION=${TARGET_EXTENSION}"
             # pull in the build artifacts
             - cp -r ${CODEBUILD_SRC_DIR_buildPipeline}/* .
-            - VERSION=$(node -e "console.log(require('./packages/toolkit/package.json').version);")
-            - UPLOAD_TARGET=$(ls aws-toolkit-vscode*.vsix)
+            - VERSION=$(node -e "console.log(require('./packages/${TARGET_EXTENSION}/package.json').version);")
+            - UPLOAD_TARGET=$(ls *.vsix)
```
Comment (justinmk3) [inline buildspec/release/50githubrelease.yml:31]: Seems fragile. But not a blocker for this PR.

File: buildspec/release/50githubrelease.yml [hunk 3]
```diff
@@ -19,22 +19,24 @@ phases:
...
             - HASH_UPLOAD_TARGET=${UPLOAD_TARGET}.sha384
             - 'HASH=$(sha384sum -b $UPLOAD_TARGET | cut -d" " -f1)'
             - echo "Writing hash to $HASH_UPLOAD_TARGET"
             - echo $HASH > $HASH_UPLOAD_TARGET
             - echo "posting $VERSION with sha384 hash $HASH to GitHub"
             - RELEASE_MESSAGE="AWS Toolkit for VS Code $VERSION"
             - |
-                if [ $STAGE = "prod" ]; then
+                if [ $STAGE = "prod" || $TARGET_EXTENSION != 'toolkit' ]; then
```
Comment (justinmk3) [inline buildspec/release/50githubrelease.yml:39]: Do we need to revisit this later (i.e. is it in our checklist or backlog)?

**Review by justinmk3 — APPROVED** [review]

File: buildspec/release/50githubrelease.yml (hunk 3 above)
Comment (hayemaxi) [reply buildspec/release/50githubrelease.yml:39]: yes

File: buildspec/release/20buildrelease.yml (hunk 1 above)
Comment (justinmk3) [reply buildspec/release/20buildrelease.yml:42]: It's harmless though, right? And slightly beneficial because it makes our intentions explicit?

**Pull request merged**